config/config.go

```diff
@@ -313,23 +318,11 @@ func loadService(v *viper.Viper) (configmodels.Service, error) {
 }
 
 // LoadReceiver loads a receiver config from v under the subkey receiverKey using the provided factories.
-func LoadReceiver(receiverKey string, v *viper.Viper, factories map[string]component.ReceiverFactoryBase) (configmodels.Receiver, error) {
+func LoadReceiver(receiverKey string, v *viper.Viper, factory component.ReceiverFactoryBase) (configmodels.Receiver, error) {
```

Comment:
1. Can we add a comment that says that the `factory` must correspond to the receiver type specified in the `receiverKey`?
2. Can we add a runtime check for it by comparing `factory.Type()==typeStr`?